server/sonar-db-dao/src/main/java/org/sonar/db/component/ComponentDao.java

```diff
@@ -321,6 +326,10 @@ public void update(DbSession session, ComponentUpdateDto component) {
     mapper(session).update(component);
   }
 
+  public void updateTags(DbSession session, ComponentDto component) {
```

Comment:
we cannot update a component like you said. There's no method to update a component except in the context of Compute Engine with the b_ columns